### ISSUE TITLE
Production app running a test app when importing Appearance Screen

### ISSUE THREAD

**Issue by williambrode** [fyne-io/fyne#4792]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When using the appearance screen in my app I realized while debugging that a test app was running.  `testapp.go` has this:

```golang
// ensure we have a dummy app loaded and ready to test
func init() {
	NewApp()
}
```

And `softwarecanvas.go` which is used by the appearance screen is importing test:

```golang
import (
	"fyne.io/fyne/v2/driver/software"
	"fyne.io/fyne/v2/test"
)
```

### How to reproduce

Run example code in the debugger and notice the stack like this:

```
runtime.gopark (proc.go:364) runtime
runtime.chanrecv (chan.go:583) runtime
runtime.chanrecv1 (chan.go:442) runtime
test.NewApp.func1 (testapp.go:166) fyne.io/fyne/v2/test
runtime.goexit (asm_amd64.s:1594) runtime
 - Async Stack Trace
test.NewApp (testapp.go:164) fyne.io/fyne/v2/test
```

### Example code

```golang
import (
   "fyne.io/fyne/v2/cmd/fyne_settings/settings"
)
```

### Fyne version

2.4.4

### Go compiler version

1.19.3

### Operating system and version

Windows 10

**Comment by andydotxyz** [comment]
I'm not certain what bug this is reporting...

(and on another note a test app does not "run", it exists in memory ready for when something is rendered).

**Comment by williambrode** [comment]
Typically production code should not reference test code.  So at least I would imagine the code should be restructured if this is all intended.  But my bigger concern was that we were accidentally running this goroutine with no benefit.  The comments seem to indicate its only used for testing.

**Comment by andydotxyz** [comment]
It's a package for helping run test apps in memory rather than actually test code - but I get your point.

This should be resolved with other work we have in the 2.5 release because `ThemeOverride` container supports this functionality without the test package.
The complete fix isn't on develop yet but it's close.

**Comment by andydotxyz** [comment]
Fixed on develop. I simplified the preview as well - I hope it still delivers
